src/components/UserSelector.tsx

```diff
@@ -38,10 +132,20 @@
     return () => {
       document.removeEventListener('click', handleDocumentClick);
     };
-    // we don't want to listening for outside clicks
-    // when the Dopdown is closed
   }, [expanded]);
 
+  if (loading) {
+    return (
+      <div data-cy="UserSelector">
+        <Loader />
+      </div>
+    );
+  }
+
+  if (error) {
+    return <div data-cy="UserSelector">{error}</div>;
```

Comment:
Ensure that the error message displayed here is user-friendly. Currently, it directly displays the error, which might not be suitable for end-users. Consider mapping errors to more descriptive messages.